cli/js/timers_test.ts

```diff
@@ -7,21 +7,22 @@ function deferred(): {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   reject: (reason?: any) => void;
 } {
-  let resolve;
-  let reject;
-  const promise = new Promise((res, rej): void => {
+  let resolve: ((value?: {} | PromiseLike<{}>) => void) | undefined = undefined;
```

Comment:
you should be able to just do `let resolve: ((value?: {} | PromiseLike<{}>) => void);`